File: DungeonsOfNoudarNDK/app/src/main/java/pt/b13h/noudar/NoudarJNI.java
package pt.b13h.noudar;

import android.content.res.AssetManager;

public class NoudarJNI {
    static {
        System.loadLibrary("native-lib");
    }

    public static native void initAssets(AssetManager assetManager);
    public static native void getPixelsFromNative(byte[] javaSide);
    public static native void sendCommand(char cmd);
    public static native int getSoundToPlay();

}

File: DungeonsOfNoudarNDK/app/src/main/java/pt/b13h/noudar/NoudarJNI.kt
package pt.b13h.noudar

import android.content.res.AssetManager

object NoudarJNI {
    init {
        System.loadLibrary("native-lib")
    }

    external fun initAssets(assetManager: AssetManager?)
    external fun getPixelsFromNative(javaSide: ByteArray?)
    external fun sendCommand(cmd: Char)
    external fun getSoundToPlay(): Int
}
